Tests/FluentAssertions.Specs/Collections/CollectionAssertionSpecs.cs

```diff
@@ -1289,7 +1262,7 @@ public void When_asserting_identical_collections_to_be_equal_it_should_enumerate
             actual.Should().Equal(expected);
 
             // Assert
-            actual.GetEnumeratorCallCount.Should().Be(1);
+            actual.GetEnumeratorCallCount.Should().BeLessOrEqualTo(1);
```

Comment:
Why did you need this?

Reply:
Historically #438 added materialization of `Subject` to avoid multiple enumerations.

in master `ConvertOrCastToCollection<object>` always did an enumeration of `Subject` (as `CountingGenericCollection<int>` doesn't implement `ICollection<object>`) using `CountingGenericCollection.GetEnumerator`. 
That gave the expected `GetEnumeratorCallCount` with value `1`.
```cs
NotEqual(IEnumerable unexpected
...
ICollection<object> actualItems = Subject.ConvertOrCastToCollection<object>();
```

In this PR `ConvertOrCastToCollection` now just casts `CountingGenericCollection<int>` to `ICollection<int>` and avoids enumeration using `CountingGenericCollection.GetEnumerator`.
Instead uses `CountingGenericCollection.CopyTo`, which does not increment `GetEnumeratorCallCount`, to make a defensive copy of `Subject`.

The other two enumerations of `actualItems` are `SequenceEquals` and maybe `EnumerableValueFormatter.Format` if failing.

I think the tests might be misleading, but it depends on our usage of materialization.
As I understand it, we materialize to avoid enumerating _lazy_ enumerables multiple times, but accept multiple enumerations of e.g. `ICollection<T>` - this is at least how `ConvertOrCastToCollection` and `ConvertOrCastToList` are implemented.

Should `When_asserting_identical_collections_to_be_equal_it_should_enumerate_the_subject_only_once` and `When_asserting_identical_collections_to_not_be_equal_it_should_enumerate_the_subject_only_once`
 rather be using a `CountingGenericLazyEnumerable`?

Reply:
I just ran that test and it returns `1` for `GetEnumeratorCallCount`, which is what I was expecting.

Reply:
I must have edited `When_asserting_identical_collections_to_be_equal_it_should_enumerate_the_subject_only_once` and then changing something else... 🤦 

If you run `When_asserting_identical_collections_to_not_be_equal_it_should_enumerate_the_subject_only_once` it is `0`.